pkg/controller/tas/node_failure_controller.go

```diff
@@ -323,3 +322,43 @@ func (r *nodeFailureReconciler) reconcileForReplaceNodeOnPodTermination(ctx cont
 		return ctrl.Result{}, patchErr
 	}
 }
+
+// removeNodeToReplaceAnnotation finds workloads with the specified node in the NodeToReplaceAnnotation
+// and removes the annotation
+func (r *nodeFailureReconciler) removeNodeToReplaceAnnotation(ctx context.Context, nodeName string, affectedWorkloads sets.Set[types.NamespacedName]) error {
+	var workloadProcessingErrors []error
+	for wlKey := range affectedWorkloads {
+		log := r.log.WithValues("workload", wlKey, "nodeName", nodeName)
+		// fetch workload.
+		var wl kueue.Workload
+		if err := r.client.Get(ctx, wlKey, &wl); err != nil {
+			if apierrors.IsNotFound(err) {
+				log.V(4).Info("Workload not found, skipping")
+			} else {
+				log.V(2).Error(err, "Failed to get workload")
+				workloadProcessingErrors = append(workloadProcessingErrors, err)
+			}
+			continue
+		}
+
+		if wl.Annotations[kueuealpha.NodeToReplaceAnnotation] != nodeName {
```

Comment:
```suggestion
		if wl.Annotations != nil &&  wl.Annotations[kueuealpha.NodeToReplaceAnnotation] != nodeName {
```

Isn't this potential null pointer?